src/progress/pct/compile.p

```diff
@@ -499,6 +545,25 @@ PROCEDURE compileXref.
 
 END PROCEDURE.
 
+PROCEDURE printErrorsWarningsJson.
+  IF ( outputType EQ 'json' ) THEN DO:
+    DEFINE VARIABLE httProjectErrors    AS HANDLE.
+    DEFINE VARIABLE httProjectWarnings  AS HANDLE.
+    
+    httProjectErrors   = TEMP-TABLE ttProjectErrors:HANDLE.
+    httProjectWarnings = TEMP-TABLE ttProjectWarnings:HANDLE.
+
+    DEFINE VARIABLE ttProjectErrorsFile AS CHARACTER NO-UNDO.
+    ASSIGN ttProjectErrorsFile = PCTDir + '/':U + 'projectErrors.json':U.
```

Comment:
This file name makes me think that it can't work with multiple compilation threads. It works when you have a single thread, but multiple threads will overwrite this file.
The correct way to do that with multi threads is to return the content of the temp-tables to the Java server socket, aggregate the result, and write the JSON file at the end of the Java process.
But your use case perhaps won't use Multi threading, so it might be easier to disable jsonOutput attribute when using multiple threads.

Reply:
> But your use case perhaps won't use Multi threading, so it might be easier to disable jsonOutput attribute when using multiple threads

I don't have much knowledge base on PCT way of doing multiple threading for compiling files. I was going through the code and i think [here](https://github.com/Riverside-Software/pct/blob/2ca951efb1f799abc4a47c2c7f36255cd728de1a/src/java/com/phenix/pct/PCTBgRun.java#L319), instead of disabling jsonOutput , we should throught error.
If I added  
public void setJsonOutput(boolean json){throw err;} in the same [code](https://github.com/Riverside-Software/pct/blob/2ca951efb1f799abc4a47c2c7f36255cd728de1a/src/java/com/phenix/pct/PCTBgRun.java#L319)
 Is it fine?
Can you suggest where should I be making changes?

Reply:
For now, I'd just discard the parameter, and log a message (jsonOutput not yet supported)

Reply:
I may need your help here to implement this, the existing code is complex to understand.
Question 1. 
What would be the right place to discard the parameter? 

In  my knowledge i think [here ](https://github.com/Riverside-Software/pct/pull/383/files#diff-2425573c3e537106a0a6695921b408c3R345)
Replace it  with  sb.append("").append(';'); 
Correct me if I am wrong?

Question  2.
What would be the right place to log the message?
In my knowledge, I think [here](https://github.com/Riverside-Software/pct/blob/d928f7aac2cede3d37e1857ab56173e3fc24ad8c/src/java/com/phenix/pct/PCTBgCompile.java#L156)

with the below code :

```
if (compAttrs.getOutputType().equalsIgnoreCase("json")) {
            log("outputType parameter is not supported on multi-threaded environment", Project.MSG_WARN); //$NON-NLS-1$
        }
```
I will put the warning message, in messages.properties file.
Thanks for your anticipation.

Reply:
Q1. Yes
Q2. Yes